src/screens/auth/verifyEmail.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react'
+import { ScrollView, StyleSheet, View } from 'react-native'
+import { Headline, Caption, Button, TextInput, Text } from 'react-native-paper'
+import { useForm, Controller } from "react-hook-form"
+import { useMutation } from "urql"
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { VERIFY_EMAIL_MUTATION } from '../../api/gql'
+import routes from '../../navigation/routes'
+import { globalStyles } from '../../utils/styles'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { getRules, InputType } from '../../utils/validation'
+
+type VerifyEmailResult = { confirmEmail: { token: string } }
+type VerifyEmailRequest = { username: string; sessionCode: string; verificationCode: string }
+
+export const VerifyEmail = ({ route, navigation }) => {
+
+    const [isLoading, setLoading] = useState(false)
+    const { control, handleSubmit, errors } = useForm()
+    const [verifyEmailResult, verifyEmail] = useMutation<VerifyEmailResult, VerifyEmailRequest>(VERIFY_EMAIL_MUTATION)
+
+    const textInput = (type: InputType, hasError: boolean) => {
+        return (
+            <Controller
+                control={control}
+                render={({ onChange, value }) => (
+                    <View style={styles.inputText}>
+                        <TextInput
+                            mode='outlined'
+                            label='Verification Code'
+                            onChangeText={value => onChange(value)}
+                            value={value}
+                            error={hasError}
+                            autoCapitalize="none" />
+                        { hasError && <Caption style={styles.error}>This is required.</Caption>}
+                    </View>
+                )}
+                name={type}
+                rules={getRules(type)}
+                defaultValue="" />
+        )
+    }
+
+    const onSubmit = (data: VerifyEmailRequest) => {
+        const { sessionCode, username } = route.params;
```

Comment:
The `sessionCode` variable here was empty when I tried it out.

Reply:
I had it commented out for testing purposes to avoid creating dummy accounts. Just added it back.